caikit_nlp/modules/text_generation/text_generation_tgis.py

```diff
@@ -333,7 +349,26 @@ def run_tokenizer(
             TokenizationResults
                 The token count
         """
+        if self._tgis_backend:
+            self._register_model_connection_with_context(context)
+
         if self._model_loaded:
             return self.tgis_generation_client.unary_tokenize(
                 text=text,
             )
+
+    def _register_model_connection_with_context(
+        self, context: Optional[RuntimeServerContextType]
+    ):
+        ok, route_info = get_route_info(context)
+        if ok:
+            log.debug(
+                "<NLP15770311D> Registering remote model connection with context override: 'hostname: %s'",
+                route_info,
+            )
+            self._tgis_backend.register_model_connection(
+                self.model_name, {"hostname": route_info}, fill_with_defaults=True
+            )
+        else:
+            self._tgis_backend.register_model_connection(self.model_name)
```

Comment:
Same comment about not needing the `else` clause

Reply:
done.